.jenkins/checks.Jenkinsfile

```
@@ -1,19 +1,34 @@
 pipeline {
-    libraries { lib 'adverity-shared-library@2.6.0'}
+    libraries { lib 'adverity-shared-library@2.10.0'}
     agent { label "PRJob" }
     environment {
         NODE_VERSION = getNodeVersion()
     }
+    options {
+        withAWS(role: 'CI', roleAccount: '221160807535')
+    }
     stages {
         stage('Preparation') {
             steps {
                 echo "Using node version: ${env.NODE_VERSION}"
             }
         }
         stage('Install dependencies') {
+            options {
+                withAWS(role: 'CI', roleAccount: '508912190628', region: 'eu-west-1')
+            }
+            environment {
+                CODEARTIFACT_AUTH_TOKEN = "${sh(script: 'aws codeartifact get-authorization-token --domain adverity --domain-owner 508912190628 --query authorizationToken --output text', returnStdout: true)}".trim()
+            }
             steps {
+                // sed is needed because there are problems inejcting the environment while using  the `nvm` closure
+                sh """
+                   sed -i "s/\\\${CODEARTIFACT_AUTH_TOKEN}/${env.CODEARTIFACT_AUTH_TOKEN}/g" .npmrc
+                """
                 nvm(env.NODE_VERSION) {
-                    sh 'npm ci'
+                    sh """
+                       npm ci
+                    """
```

any problems with this?

```suggestion
                    sh 'npm ci'
```

---

https://github.com/pmedianetwork/eslint-config/pull/87/commits/15a1f96770224aad18b8f71696e2844b3f1641e1